parens_2.py

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env python
+
+import sys
+
+
+def _parens(prefix, left, right):
+    if left == 0 and right == 0: sys.stdout.write(prefix)
+    if left > 0: _parens(''.join([prefix, '(']), left-1, right)
```

Comment:
Understand... perhaps it was a false positive in my tests here.
Will check that out again... 

In terms of the join method, it states that could be a much faster string manipulation that concatenation using the + (plus) operator.

Will check again this functionality and my cython version as well, since it uses the sys.stdout.write as mentioned above.